Entities/Common/GUI/ActorHUDStartPos.as

```diff
@@ -14,12 +16,91 @@ f32 getHUDY()
 const f32 HUD_X = getHUDX();
 const f32 HUD_Y = getHUDY();
 
+bool hoverOnResupplyIcon(Vec2f icon_pos, Vec2f icon_size)
+{
+	Vec2f mouse_pos = getControls().getMouseScreenPos();
+
+	if (mouse_pos.x > icon_pos.x && mouse_pos.x < icon_pos.x + icon_size.x * 2
+		&& mouse_pos.y > icon_pos.y && mouse_pos.y < icon_pos.y + icon_size.y * 2)
+	{
+		return true;
+	}
+
+	return false;
+}
+
 Vec2f getActorHUDStartPosition(CBlob@ blob, const u8 bar_width_in_slots)
 {
 	f32 width = bar_width_in_slots * 40.0f;
 	return Vec2f(getHUDX() + 180 + 50 + 8 - width, getHUDY() - 40);
 }
 
+void DrawResupplyOnHUD(CBlob@ this, Vec2f tl)
+{
+	CPlayer@ p = this.getPlayer();
+	if (p is null) return;
+	
+	string name = this.getName();
+
+	GUI::SetFont("menu");
+
+	string propname = getCTFTimerPropertyName(p, "builder");
+
+	if (getRules().exists(propname)) 
+	{
+		int wood_amount = matchtime_wood_amount;
+		int stone_amount = matchtime_stone_amount;
+		if (getRules().isWarmup())
+		{
+			wood_amount = warmup_wood_amount;
+			stone_amount = warmup_stone_amount;
+		}
+
+		s32 next_items = getRules().get_s32(propname);
+
+		u32 secs = ((next_items - 1 - getGameTime()) / getTicksASecond()) + 1;
+		string units = ((secs != 1) ? " seconds" : " second");
+
+		string resupply_available = getTranslatedString("Go to a builder shop or a respawn point to get a resupply of {WOOD} wood and {STONE} stone.")
+			.replace("{WOOD}", "" + wood_amount)
+			.replace("{STONE}", "" + stone_amount);
+
+		string resupply_unavailable = getTranslatedString("Next resupply of {WOOD} wood and {STONE} stone in {SEC}{TIMESUFFIX}.")
+			.replace("{SEC}", "" + secs)
+			.replace("{TIMESUFFIX}", getTranslatedString(units))
+			.replace("{WOOD}", "" + wood_amount)
+			.replace("{STONE}", "" + stone_amount);
+
+		string short_secs = secs + "s";
+
+		Vec2f icon_pos = tl;
+		Vec2f icon_size = Vec2f(16, 16);
+
+		bool hover = hoverOnResupplyIcon(icon_pos, icon_size);
+
+		if (next_items > getGameTime())
+		{
+			GUI::DrawIcon("Entities/Common/GUI/ResupplyIcon.png", 0, icon_size, icon_pos, 1.0f);
+			GUI::DrawTextCentered(short_secs, icon_pos + Vec2f(16, 38), color_white);
+
+			if (hover)
+			{
+				GUI::DrawTextCentered(resupply_unavailable, icon_pos - Vec2f(0, 24), color_white);
+			}
+		}
+		else
+		{
+			GUI::DrawIcon("Entities/Common/GUI/ResupplyIcon.png", 1, icon_size, icon_pos, 1.0f);
+
```

Comment:
Looks a bit empty when the resupplies are ready. It might not be very necessary but I think it would be good to have some short text/icon to hint where you would get the resupplies here?

Reply:
i don't think there's enough place for it
i just changed it so that there's a small offset that moves it to the down when it's ready